### DISCUSSION TITLE
NATS python publish credentials authorization

### DISCUSSION THREAD

**Discussion by pturanyi** [nats-io/nats.py#207] [Q&A]
Hello everybody.
Please can you provide how to pass token to NATS server using nats-pub.py script.  
`python /tmp/nats-pub.py --creds ? -s nats://eventbus-default-stan-svc:4222 {label} -d '{json}'`    

I can get NATS access configuration
```
"config": {
    "nats": {
        "accessSecret": {
            "key": "client-auth",
            "name": "eventbus-default-client"
        },
        "auth": "token",
        "clusterID": "eventbus-default",
        "url": "nats://eventbus-default-stan-svc:4222"
    }
}
```
How to configure request to pass authorization because now I get Error: nats: 'Authorization Violation'.

**Comment by roguh** [comment]
Hello, you'll need to pass the `token` keyword argument to the connect function. Simply modify nats-pub.py in line https://github.com/nats-io/nats.py/blob/f011db7611bc9a3ed998856442e42ffce147f0e1/examples/nats-pub/__main__.py#L57 by adding `"token": <your token here>`

**Reply by pturanyi** [reply]
Thank you for answer. It is shame that something basic as authentication needs change of script and is not used as input parameter :(
